knowledge-base/gantt-different-colors-for-tasks.md

```diff
@@ -0,0 +1,230 @@
+                    Color = Colors[i]
+                };
+
+            Data.Add(newItem);
+            var parentId = LastId;
+            LastId++;
+
+            for (int j = 0; j < 5; j++)
+            {
+                Data.Add(new FlatModel()
+                    {
+                        Id = LastId,
+                        ParentId = parentId,
+                        Title = "    Task " + i + " : " + j.ToString(),
+                        Start = new DateTime(2021, 7, 5 + j),
+                        End = new DateTime(2021, 7, 6 + i + j),
+                        PercentComplete = Math.Round(random.NextDouble(), 2),
+                        Assignee = "    Employee " + j,
+                        Color = Colors[j]
+                    });
+
+                LastId++;
+            }
+        }
+
+        base.OnInitialized();
+    }
+
+    private void UpdateItem(GanttUpdateEventArgs args)
+    {
+        var item = args.Item as FlatModel;
+
+        var foundItem = Data.FirstOrDefault(i => i.Id.Equals(item.Id));
+
+        if (foundItem != null)
+        {
+            foundItem.Title = item.Title;
+            foundItem.Start = item.Start;
+            foundItem.End = item.End;
+            foundItem.PercentComplete = item.PercentComplete;
+        }
+    }
+
+    private void DeleteItem(GanttDeleteEventArgs args)
+    {
+        var item = Data.FirstOrDefault(i => i.Id.Equals((args.Item as FlatModel).Id));
+
+        RemoveChildRecursive(item);
+    }
+
+    private void RemoveChildRecursive(FlatModel item)
+    {
+        var children = Data.Where(i => item.Id.Equals(i.ParentId)).ToList();
+
+        foreach (var child in children)
+        {
+            RemoveChildRecursive(child);
+        }
+
+        Data.Remove(item);
+    }
+
+    class FlatModel
+    {
+        public int Id { get; set; }
+        public int? ParentId { get; set; }
+        public string Title { get; set; }
+        public double PercentComplete { get; set; }
+        public DateTime Start { get; set; }
+        public DateTime End { get; set; }
+        public string Assignee { get; set; }
+        public string Color { get; set; }
+    }
+}
+
+<style>
+    .k-task-single {
+        background-color: none;
+    }
+
+    .k-task-actions {
+        position: absolute;
+        right: 0;
+        top: 14px;
+    }
+
+    .k-task-template {
+        padding: 0px;
+    }
+
+    .k-task-content {
+        display: block;
+    }
+
+    .wrapper {
+        padding-top: 4px;
+        padding-bottom: 4px;
+        padding-left: 8px;
+        padding-right: 8px;
+        border-radius: 4px
+    }
+
+    .task-title {
+        font-size: 18px;
+        font-weight: bold;
+        border-bottom: 1px solid;
+    }
+
+    .task-assignee {
+        font-style: italic;
+    }
+
+    .progress {
+        position: absolute;
+        left: 0;
+        bottom: 0;
+        height: 4px;
+        background: rgba(252, 98, 87, .7);
+    }
+</style>
+````
```

Comment:
Add a `## See Also` section with links to:

* Gantt overview
* Gantt live demo
* Gantt task template
* Gantt API

In addition, for SEO purposes you might want to figure out a Gantt page from which to link to this KB.